File: LeetCodeSolutions/LeetCode_1232.py
class Solution:
    def checkStraightLine(self, coordinates: List[List[int]]) -> bool:
        if len(coordinates) < 3:
            return True
        p0, p1 = coordinates[0], coordinates[1]

        def helper(p):
            return (p[0] - p0[0]) * (p1[1] - p0[1]) == (p[1] - p0[1]) * (p1[0] - p0[0])

        for p in coordinates[2:]:
            if not helper(p):
                return False
        return True

File: LeetCodeSolutions/LeetCode_1232.cpp
class Solution {
public:
    bool checkStraightLine(vector<vector<int>>& coordinates) {
        int x1 = coordinates[0][0], y1 = coordinates[0][1], x2 = coordinates[1][0], y2 = coordinates[1][1];
        for(int i = 2; i < coordinates.size(); ++i){
            int x = coordinates[i][0], y = coordinates[i][1];
            if ((y2 - y1) * (x - x1) == (y - y1) * (x2 - x1)) continue;
            return false;
        }
        return true;
    }
};